File: samples/OpenIdRelyingPartyWebForms/Code/State.cs
namespace OpenIdRelyingPartyWebForms {
	using System.Web;
	using DotNetOpenAuth.OpenId.Extensions.AttributeExchange;
	using DotNetOpenAuth.OpenId.Extensions.ProviderAuthenticationPolicy;
	using DotNetOpenAuth.OpenId.Extensions.SimpleRegistration;

	/// <summary>
	/// Strong-typed bag of session state.
	/// </summary>
	public class State {
		public static ClaimsResponse ProfileFields {
			get { return HttpContext.Current.Session["ProfileFields"] as ClaimsResponse; }
			set { HttpContext.Current.Session["ProfileFields"] = value; }
		}

		public static FetchResponse FetchResponse {
			get { return HttpContext.Current.Session["FetchResponse"] as FetchResponse; }
			set { HttpContext.Current.Session["FetchResponse"] = value; }
		}

		public static string FriendlyLoginName {
			get { return HttpContext.Current.Session["FriendlyUsername"] as string; }
			set { HttpContext.Current.Session["FriendlyUsername"] = value; }
		}

		public static PolicyResponse PapePolicies {
			get { return HttpContext.Current.Session["PapePolicies"] as PolicyResponse; }
			set { HttpContext.Current.Session["PapePolicies"] = value; }
		}

		public static string GoogleAccessToken {
			get { return HttpContext.Current.Session["GoogleAccessToken"] as string; }
			set { HttpContext.Current.Session["GoogleAccessToken"] = value; }
		}

		public static void Clear() {
			ProfileFields = null;
			FetchResponse = null;
			FriendlyLoginName = null;
			PapePolicies = null;
			GoogleAccessToken = null;
		}
	}
}

File: samples/OpenIdRelyingPartyWebFormsVB/Code/State.vb
Imports DotNetOpenAuth.OpenId.Extensions.ProviderAuthenticationPolicy
Imports DotNetOpenAuth.OpenId.Extensions.SimpleRegistration
Imports DotNetOpenAuth.OpenId.Extensions.AttributeExchange

Public Class State
	Public Shared Property ProfileFields() As ClaimsResponse
		Get
			Return HttpContext.Current.Session("ProfileFields")
		End Get
		Set(ByVal value As ClaimsResponse)
			HttpContext.Current.Session("ProfileFields") = value
		End Set
	End Property

	Public Shared Property FetchResponse() As FetchResponse
		Get
			Return HttpContext.Current.Session("FetchResponse")
		End Get
		Set(ByVal value As FetchResponse)
			HttpContext.Current.Session("FetchResponse") = value
		End Set
	End Property

	Public Shared Property FriendlyLoginName() As String
		Get
			Return HttpContext.Current.Session("FriendlyLoginName")
		End Get
		Set(ByVal value As String)
			HttpContext.Current.Session("FriendlyLoginName") = value
		End Set
	End Property

	Public Shared Property PapePolicies() As PolicyResponse
		Get
			Return HttpContext.Current.Session("PapePolicies")
		End Get
		Set(ByVal value As PolicyResponse)
			HttpContext.Current.Session("PapePolicies") = value
		End Set
	End Property

	Public Shared Sub Clear()
		FriendlyLoginName = Nothing
	End Sub

End Class
